### PR TITLE
Add route tables and subnets to Cloud Formation

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#2023

Sorry this PR got a bit big. At first I tried to just migrate the private route table and subnet. But I couldn't get the public route table and subnet to delete reliably in the legacy resource.

### CODE REVIEW

File: service/resource/legacyv2/adapter/adapter.go [hunk 1]
```diff
@@ -37,15 +37,16 @@ type Adapter struct {
 	ASGType          string
 	AvailabilityZone string
 	ClusterID        string
+	VPCID            string
```
Comment (rossf7) [inline service/resource/legacyv2/adapter/adapter.go]: I moved VPCID to the main adapter as its needed by multiple adapters.

File: service/resource/legacyv2/adapter/internet_gateway.go [hunk 2]
```diff
@@ -1,37 +0,0 @@
-package adapter
-
```
Comment (rossf7) [inline service/resource/legacyv2/adapter/internet_gateway.go]: This adapter no longer has any fields so I removed it.

File: service/resource/legacyv2/adapter/nat_gateway.go [hunk 3]
```diff
@@ -1,39 +0,0 @@
-package adapter
```
Comment (rossf7) [inline service/resource/legacyv2/adapter/nat_gateway.go]: Same with the NAT gateway adapter.

File: service/resource/legacyv2/adapter/adapter.go [hunk 4]
```diff
@@ -37,15 +37,16 @@ type Adapter struct {
...
 
 	autoScalingGroupAdapter
 	iamPoliciesAdapter
 	instanceAdapter
 	launchConfigAdapter
 	loadBalancersAdapter
-	internetGatewayAdapter
```
Comment (fgimenez) [inline service/resource/legacyv2/adapter/adapter.go]: Very nice that we no longer need these adapters! :+1:

File: service/templates/cloudformation/auto_scaling_group.yaml [hunk 5]
```diff
@@ -3,7 +3,7 @@
     Type: "AWS::AutoScaling::AutoScalingGroup"
     Properties:
       VPCZoneIdentifier:
-        - {{ .WorkerSubnetID }}
```
Comment (fgimenez) [inline service/templates/cloudformation/auto_scaling_group.yaml]: I really like these data actions changed to CF refs :heart:

**Pull request merged**